### PR TITLE
🌱 e2e/apiexportendpointslice: use SharedKcpServer

### PR DESCRIPTION
**Pull request by p0lyn0mial**
## Summary
it is possible to use a shared server because the
scheduler skips shards annotated with `experimental.core.kcp.io/unschedulable`

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by fgiloux** [comment]
This looks good to me. I just have a couple of points:

Can you please amend the comment here?
https://github.com/kcp-dev/kcp/blob/73cc8efef729522a1dc30f1e7318559982817460/test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go#L268-L270

This could be a separate test function:
https://github.com/kcp-dev/kcp/blob/73cc8efef729522a1dc30f1e7318559982817460/test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go#L342-L382
It was not the case previously due to the cost of setting up the private environment

**Comment by p0lyn0mial** [comment]
Updated the comment. IMO there is no need for extracting logic to a helper function. I like when tests are self-contained and self-descriptive.

**Comment by fgiloux** [comment]
> Updated the comment. IMO there is no need for extracting logic to a helper function. I like when tests are self-contained and self-descriptive.

I did not mean a helper function. I meant a completely separate test function as it is not directly related to the tests part of the same function. It is not critical, just thinking it may be nicer.

File: test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go [hunk 1]
```diff
@@ -352,12 +356,28 @@ func TestAPIExportEndpointSliceWithPartitionPrivate(t *testing.T) {
...
+					}
+				}
+				if !found {
+					return false, fmt.Sprintf("didn't find matching shard for %v endpoint, shards: %#v", apiExportEndpoint.URL, shards.Items)
+				}
+			}
 			return true, ""
 		}
-		return false, fmt.Sprintf("expected 1 endpoint, but got: %#v", sliceWithAll.Status.APIExportEndpoints)
+		return false, fmt.Sprintf("expected %d endpoint, but got: %#v", len(shards.Items), sliceWithAll.Status.APIExportEndpoints)
```
Comment (fgiloux) [inline test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go:381]: I am concerned that you may be introducing a flake here: What does guarantee no shard was created by a parallel test case between your call to list shards and the one to get the APIExportEndpointSlices?

Comment (p0lyn0mial) [reply test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go:381]: good point, assuming schedulable shards won't be removed we could change this logic to expect `>=` `initial shards+ shards added by this tests`.

**Comment by p0lyn0mial** [comment]
> I did not mean a helper function. I meant a completely separate test function as it is not directly related to the tests part of the same function. It is not critical, just thinking it may be nicer.

ah, this could be a default (happy-path) scenario I always wanted, yeah I can do that :)

**Pull request closed without merging**